neuralspot/ns-utils/tests/ns_free_tests.c

```diff
@@ -57,18 +62,16 @@ void ns_free_test_memory_fragmentation() {
 
     // Allocate memory in a pattern that can cause fragmentation
     for (i = 0; i < 1000; ++i) {
-        allocs[i] = ns_malloc(1000);
+        allocs[i] = ns_malloc(10);
         TEST_ASSERT_NOT_NULL(allocs[i]);
     }
 
     // Free every other allocation
     for (i = 1; i < 1000; i += 2) {
         ns_free(allocs[i]);
     }
-
     // Allocate a large block after inducing fragmentation
-    void *large_alloc = ns_malloc(1000 * 1000);
-
+    void *large_alloc = ns_malloc(1000);
```

Comment:
A quick note (no changes needed if the test passes): there are several implementations of malloc which optimize for different behaviors. We are using heap4, which trades speed for garbage collection overhead. It is designed for static use of malloc (malloc mostly at the beginning of execution, and free mostly at the end) and handles fragmentation poorly.